pkg/scheduler/objects/application.go

```diff
@@ -1531,7 +1528,7 @@ func (sa *Application) removeAllocationInternal(uuid string) *Allocation {
 	} else {
 		sa.allocatedResource = resources.Sub(sa.allocatedResource, alloc.GetAllocatedResource())
 		// When the resource trackers are zero we should not expect anything to come in later.
-		if resources.IsZero(sa.pending) && resources.IsZero(sa.allocatedResource) {
+		if sa.hasZeroAllocations() {
 			if err := sa.HandleApplicationEvent(CompleteApplication); err != nil {
```

Comment:
Can we make this as common block for both ph and normal alloc something like given below with var event being set with different values based on the various condition in the above code block (may be you can replace different if's with switch statement) ?

	// When the resource trackers are zero we should not expect anything to come in later.
	if sa.hasZeroAllocations() {
		event = CompleteApplication
	}

	if err := sa.HandleApplicationEvent(event); err != nil {
		log.Logger().Warn("Application state not changed while removing a placeholder allocation",
			zap.String("currentState", sa.CurrentState()),
			zap.String("event", event.String()),
			zap.Error(err))
	}